src/lib/classes/QScrollObserver.svelte.ts

```diff
@@ -0,0 +1,109 @@
+type Direction = "up" | "right" | "down" | "left";
+
+type WindowScrollType = "scrollX" | "scrollY";
+type DocumentScrollType = "scrollLeft" | "scrollTop";
+
+type ScrollType = {
+  window: WindowScrollType;
+  document: DocumentScrollType;
+};
+
+/**
+ * A scroll observer utility.
+ * @example
+ * ```js
+ * // Inside the <script> tag
+ * const scroll = new QScrollObserver();
+ *
+ * $inspect(scroll.direction)
+ * ```
+ *
+ * @param horizontal - If set to `true`, observe horizontal scrolling rather than vertical
+ * @param debounce - Time in milliseconds between each observation update
+ */
+export default class QScrollObserver {
+  private scrollType: ScrollType = {
```

Comment:
Can we, as a rule of thumb, please use `protected` instead of `private`? It will make inheriting much easier.

Reply:
I don't see why this class should be inherited from but sure ig...